### PR TITLE
ESLint Standard

### PR DESCRIPTION
**Pull request by zcei**
Umbrella issue: #1333 

Use a unified ESLint config instead of maintaining our own set of rules.

`no-var` & `prefer-const` has been added, so we get ESLint autofixes for free 🙂 

Only the first commit needs to be reviewed, as the rest can be considered safe, as the transforms for autofixes are tested by ESLint & the community.

### CODE REVIEW

**Comment by axe312ger** [comment]
A few manual fixes are still needed:
```
/home/travis/build/axios/axios/lib/adapters/http.js
   34:16  error    'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
   36:16  error    'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  114:24  error    'new Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  132:5   warning  Unexpected var, use let or const instead                                                                                                                no-var
/home/travis/build/axios/axios/lib/helpers/btoa.js
  19:5  warning  Unexpected var, use let or const instead  no-var
/home/travis/build/axios/axios/lib/helpers/isAbsoluteURL.js
  13:24  error  Unnecessary escape character: \+  no-useless-escape
  13:28  error  Unnecessary escape character: \.  no-useless-escape
/home/travis/build/axios/axios/lib/helpers/isURLSameOrigin.js
  47:7  error  'originURL' is never reassigned. Use 'const' instead  prefer-const
/home/travis/build/axios/axios/lib/utils.js
  218:7  error  unnecessary '.call()'  no-useless-call
  224:9  error  unnecessary '.call()'  no-useless-call
✖ 10 problems (8 errors, 2 warnings)
```

**Comment by zcei** [comment]
These are most likely to be resolved in the future PRs. This is only about auto-fixable stuff for a good foundation, as there is no manual review needed.

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "browser": "NODE_ENV=production webpack",
     "browser:dev": "webpack",
     "build": "npm run clean && npm run cjs && npm run browser",
+    "lint": "eslint lib test",
```
Comment (Khaledgarbaya) [inline package.json]: It would be nice to have another npm script `format` so future contributor can use it to reformat their code before submitting a PR

Comment (Khaledgarbaya) [reply package.json]: in a second thought this might pollute the changes in the PR, adding a ci step to lint would be better I guess

Comment (zcei) [reply package.json]: Added linting to test script 🙂

**Pull request merged**